src/Model/ModelTrait.php

```diff
@@ -140,13 +145,16 @@ private function getFirstParameter($method)
      * @param \ReflectionParameter $param
      * @param mixed $value
      * @param bool $isCollection
-     * @return object
+     * @return mixed
+     * @throws ModelException
+     * @throws \ReflectionException
      */
-    private function constructClasses(\ReflectionParameter $param, $value, $isCollection = false)
+    private function constructClasses(\ReflectionParameter $param, mixed $value, bool $isCollection = false): mixed
     {
-        $class = $param->getClass();
-        if (!empty($class)) {
+        $class = $param->getType() && !$param->getType()->isBuiltin()
```

Comment:
This gets more complex now that union types are a thing. For now, I think we need to check that `$param->getType()` returns a `ReflectionNamedType` object before checking `isBuiltIn()`

Might be worth setting `param->getType()` to a variable since we use it a lot

Reply:
Fixed

Reply:
Not quite. `isBuiltIn` is a method of `ReflectionNamedType` and you use that method before checking the class

Reply:
Also, if you're checking the type (with `instanceof` (hint)) you don't need the boolean-ey check at the start

Reply:
Should be fixed now, switched to instanceof